packages/components/src/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,299 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React, { ButtonHTMLAttributes, AnchorHTMLAttributes } from 'react';
+import styled, { css } from 'styled-components';
+import {
+  space,
+  margin,
+  width,
+  lineHeight,
+  MarginProps,
+  WidthProps,
+  variant,
+} from 'styled-system';
+import { composeTextStyle, Theme, ThemeProps } from '@t3n/theme';
+import { Loader } from '../Loader';
+import { Icon } from '../Icon';
+
+export type IconButtonAsType = 'button' | 'a';
+export type IconButtonVariant = 'primary' | 'secondary';
+export type IconButtonColorVariant = 'default' | 'inverse' | 'highlight';
+export type IconButtonSizeVariant = 'small' | 'regular' | 'big';
+
+export interface IconButtonProps
+  extends ButtonHTMLAttributes<any>,
```

Comment:
Für die Types ist der Tip, den Stolle vorhin geschickt hat ideal, um je nach Typ (button/link) die richtigen Props zu definieren.

Reply:
Also, ich hab mir das jetzt ne Weile angeguckt und bin mir nicht sicher wie sinnvoll das hier ist. Du meinst, man könnte dann je nachdem, ob es ein <a> oder ein <button> ist, könnte man dann entweder die ButtonHTMLAttributes oder die AnchorHTMLAttributes extenden? Alles andere ist ja in jeder Variation möglich. Das könnte ich dann aber auch nicht so schick inline schreiben, wie in diesem Tipp, denn ich muss ja ein interface schreiben, um ein interface zu extenden. Das sieht dann so aus und ist meiner Meinung nach zwar aufgeräumt, aber schwerer nachzuvollziehen, oder?

`
export type IconButtonAsType = "button" | "a";
export type IconButtonVariant = "primary" | "secondary";
export type IconButtonColorVariant = "default" | "inverse" | "highlight";
export type IconButtonSizeVariant = "small" | "regular" | "big";

interface IconButtonBaseProps extends MarginProps, WidthProps {
  size?: IconButtonSizeVariant;
  variant?: IconButtonVariant;
  colorVariant?: IconButtonColorVariant;
  icon: React.FC<React.SVGProps<SVGSVGElement>>;
  label?: string;
  loading?: boolean;
  expanded?: boolean;
}

interface IconButtonButtonTypeProps
  extends IconButtonBaseProps,
    ButtonHTMLAttributes<any> {
  as: "button";
}

interface IconButtonATypeProps
  extends IconButtonBaseProps,
    Omit<AnchorHTMLAttributes<any>, "type"> {
  as: "a";
}

export type IconButtonProps = IconButtonATypeProps | IconButtonButtonTypeProps;
`

Reply:
Also ich habe das jetzt gerade mal eingebaut, und ich muss deswegen ungefähr alles super verkomplizieren, weiß nicht, ob das hier wirklich ideal ist 🙈

Reply:
Hab das jetzt mal gepusht, der letzte commit ists. Schau gerne mal rüber, ob wir das so lassen wollen.

Reply:
Finde ich fine, vor allem weil wir dadurch korrektere Typings haben und nur noch die Props des jeweiligen Elements haben.